WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/WooShipping Create Shipping Labels/WooShippingAddCustomPackageViewModelTests.swift

```diff
@@ -0,0 +1,158 @@
+import XCTest
+@testable import WooCommerce
+import Yosemite
+
+final class WooShippingAddCustomPackageViewModelTests: XCTestCase {
+    func test_it_inits_with_empty_field_values() {
+        // Given/When
+        let viewModel = WooShippingAddCustomPackageViewModel()
+
+        // Then
+        XCTAssertNotNil(viewModel)
+        XCTAssertEqual(viewModel.fieldValues.isEmpty, true)
+        XCTAssertEqual(viewModel.packageType, WooShippingPackageType.box)
+        XCTAssertEqual(viewModel.showSaveTemplate, false)
+        XCTAssertEqual(viewModel.packageTemplateName, "")
+        XCTAssertEqual(viewModel.areFieldValuesInvalid, true)
+    }
+
+    func test_clear_field_values() {
+        // Given
+        let viewModel = WooShippingAddCustomPackageViewModel()
+
+        // When
+        viewModel.clearFieldValues()
+
+        // Then
+        XCTAssertNotNil(viewModel)
+        XCTAssertEqual(viewModel.fieldValues.isEmpty, true)
+        XCTAssertEqual(viewModel.areFieldValuesInvalid, true)
+    }
+
+    func test_reset_values() {
+        // Given
+        let viewModel = WooShippingAddCustomPackageViewModel()
+
+        // When
+        viewModel.fillWithDummyFieldValues()
+        viewModel.showSaveTemplate = true
+        viewModel.packageTemplateName = "a"
+        viewModel.resetValues()
+
+        // Then
+        XCTAssertNotNil(viewModel)
+        XCTAssertEqual(viewModel.fieldValues.isEmpty, true)
+        XCTAssertEqual(viewModel.areFieldValuesInvalid, true)
+        XCTAssertEqual(viewModel.showSaveTemplate, false)
+        XCTAssertEqual(viewModel.packageTemplateName, "")
+    }
+
+    func test_it_with_not_all_field_values_set() {
+        // Given
+        let viewModel = WooShippingAddCustomPackageViewModel()
+
+        // When
+        viewModel.clearFieldValues()
```

Comment:
Do you need this step in the test? If it inits with empty fields I'd think we could skip this and just do:

```
// When
viewModel.fieldValues[.height] = "1"
```